incubator/declarative-linking/src/main/java/org/glassfish/jersey/linking/EntityDescriptor.java

```diff
@@ -190,4 +212,44 @@ private void findLinkHeaders(Class<?> entityClass) {
             findLinkHeaders(ic);
         }
     }
+    
+    /**
+     * Find and cache the methods of the supplied class and its superclasses and
+     * interfaces.
+     * @param entityClass the class
+     */
+    private void findMethods(Class<?> entityClass) {
+        for (Method m: entityClass.getMethods()) {
+            InjectLink a = m.getAnnotation(InjectLink.class);
```

Comment:
I just spent some time looking at this to see if it was achievable and believe the code would be easier to maintain and more readable to leave as is. Each iterates over a separate collection as well and there is not a common superclass/interface for Method and Field that contains the information needed. For methods the return type is evaluated, for fields it's simply the type, for example. So a helper method would be filled with if statements and would become difficult to read/maintain I believe.